Server/Levels/StageTimer.cpp

```diff
@@ -0,0 +1,67 @@
+//
+// Created by idhant on 26-02-2024.
+//
+
+#include "StageTimer.h"
+
+StageTimer::StageTimer() {
+    std::cout << "Timer initialized" << std::endl;
+    currentState = StageTimer::Paused;
+    currentMedal = Medal::Default;
+    elapsedTime = 0.0f;
+    finalTime = 0.0f;
+}
+
+void StageTimer::Update(float dt) {
+    switch (currentState) {
+        case StageTimer::Running: {
+            elapsedTime += dt;
+            break;
+        }
+        case StageTimer::Paused: {
+            break;
+        }
+    }
+}
+
+
+void StageTimer::EndTimer() {
+    if(currentState != StageTimer::Paused) { currentState = StageTimer::Paused; }
+    CalculateMedal();
+//    CalculateScore();
+//    ResetTimer();
+}
+
+void StageTimer::ResetTimer() {
+    elapsedTime = 0.0f;
+    currentState = StageTimer::Paused;
+
+}
+
+void StageTimer::CalculateMedal() {
```

Comment:
The medal values in this function should really be variables, if we intend to have different times for medals per level.